src/OrbitTest/main_legacy_api.cpp

```diff
@@ -0,0 +1,132 @@
+// Copyright (c) 2020 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include <stdint.h>
+
+#include <algorithm>
+#include <chrono>
+#include <cmath>
+#include <cstdint>
+#include <iostream>
+#include <mutex>
+#include <string>
+#include <thread>
+#include <vector>
+
+#include "../Orbit.h"
+#include "absl/strings/str_format.h"
+
+// AsyncScopeTester starts async scopes in one thread and stops them in another.
+class AsyncScopeTester {
+ public:
+  void Start(const char* name) {
+    static uint64_t id = 0;
+    ORBIT_START_ASYNC(name, ++id);
+    std::lock_guard lock(mutex_);
+    async_scope_ids_to_stop_.push_back(id);
+  }
+
+  void Stop() {
+    std::lock_guard lock(mutex_);
+    for (uint64_t id : async_scope_ids_to_stop_) {
+      std::string str = absl::StrFormat(
+          "This is a very long dynamic string: The quick brown fox jumps over the lazy dog. This "
+          "string is associated with task id %u.",
+          id);
+      ORBIT_ASYNC_STRING(str.c_str(), id);
+      ORBIT_STOP_ASYNC(id);
+      std::this_thread::sleep_for(std::chrono::milliseconds(2));
```

Comment:
Nit: I would add a comment to indicate what the purpose of the sleep here is. Is this for throttling the amount of events caused by this code here?

Reply:
Yep, good, point, added a comment, thanks.